iothub_client/inc/iothub_client_core_common.h

```diff
@@ -234,6 +243,73 @@ extern "C"
+    * @param[in]    size                 Number of bytes of @p payload.
+    * @param[in]    payloadContentType   Payload type of @p payload.  This currenly will ALWAYS be "application/json" and applications using the default
+    *                                    options on the IoT Hub client can safely assume they will only receive JSON type.
+    *                                    Future versions of the IoT Hub client may add options that allow alternate contentTypes to be sent via
+    *                                    commands.  This will NOT be the default behavior but will require specific application opt-in.
+    * @param[out]   response             Response generated by the callback that will be sent to IoT Hub.  This must be allocated using malloc().  
+    *                                    The SDK takes ownership of the buffer on callback return and will free() it when the SDK is done with it.
+    * @param[out]   responseSize         Number of bytes application is returning in response.
+    * @param[in]    userContextCallback  User context pointer set in initial call to IoTHubDeviceClient_LL_SubscribeToCommands.
+    *
+    * @remarks
+    *            The application behavior is undefined if the user calls
+    *            the IoTHubDeviceClient_LL_Destroy function from within any callback.
+    *
+    * @return    IOTHUB_CLIENT_OK upon success or an error code upon failure.
+    */
+    typedef int(*IOTHUB_CLIENT_COMMAND_CALLBACK_ASYNC)(
+                    const char* componentName,
+                    const char* commandName,
+                    const unsigned char* payload,              
+                    size_t size,
+                    const char* payloadContentType,
+                    unsigned char** response,
+                    size_t* responseSize,
+                    void* userContextCallback);
+
+    /**
+    * @brief    Function callback application implements to receive notifications when IoT Hub acknowledges telemetry.
+    *
+    * @param    result                Result of operation sending to IoT Hub.
+    * @param    userContextCallback   Optional user specified context set in call that initiated the message send (for example, 
+    *                                 IoTHubDeviceClient_LL_SendTelemetryAsync.
+    */
+    typedef void(*IOTHUB_CLIENT_TELEMETRY_CALLBACK)(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback);
+    
+    /**
+    * @brief    Function callback application implements to receive acknowledgements of properties sent from the device to IoT Hub.  
+    *
+    * @param[in]    statusCode            Status code IoT Hub returns when receiving a property or writable property response from the device.  
+    *                                     This corresponds to HTTP status codes.
+    * @param[in]    userContextCallback   Optional user specified context set in the call that sent the properties (for example,
```

Comment:
nit: spacing between @param[in] and param name is different here than on lines 301-304.  Just noting inconsistencies if you care or have time look at. Otherwise don't worry about it.  Would it be worth us getting a clang-format file setup in the CSDK like we have in Emb C SDK so it can be automated?